ndk/src/media_error.rs

```diff
@@ -1,9 +1,22 @@
-use super::Result;
+//! Bindings for NDK media status codes.
+//!
+//! Also used outside of `libmediandk.so` in `libmidi.so` for example.
+
+#![cfg(feature = "media")]
+// The cfg(feature) bounds for some pub(crate) fn uses are non-trivial and will become even more
+// complex going forward.  Allow them to be unused when compiling with certain feature combinations.
+#![allow(dead_code)]
```

Comment:
A good example of this is when `AMidi` comes online in #353. `construct_never_null()` is used by `media+api-level-24`, but unused by `midi` (implicitly enables `api-level-24`, hence `api-level-29`). Adding a `#[cfg(all(feature = "media", feature = "api-level-24"))]` to that might work for now, but will be complicated to maintain in the future as we do not test any of these feature combinations... and only to satisfy a lint!